Src/GoogleApis.Auth.DotNet4/OAuth2/DefaultCredentialProvider.cs

```diff
@@ -0,0 +1,255 @@
+            switch (clientCredentialParameters.CredentialType)
+            {
+                case ClientCredentialType.UserCredential:
+                    try
+                    {
+                        var initializer = new GoogleAuthorizationCodeFlow.Initializer
+                        {
+                            ClientSecrets = clientCredentialParameters.ClientSecrets
+                        };
+
+                        TokenResponse token = new TokenResponse()
+                        {
+                            RefreshToken = clientCredentialParameters.RefreshToken
+                        };
+
+                        var flow = new GoogleAuthorizationCodeFlow(initializer);
+                        return new UserCredential(flow, "ApplicationDefaultCredentials", token);
+                    }
+                    catch (IOException e)
+                    {
+                        throw new InvalidOperationException("Error parsing stream contents.", e);
+                    }
+
+                case ClientCredentialType.ServiceAccountCredential:
+                    try
+                    {
+                        var initializer = new ServiceAccountCredential.Initializer(clientCredentialParameters);
+                        return new ServiceAccountCredential(initializer);
+                    }
+                    catch (IOException e)
+                    {
+                        throw new InvalidOperationException("Error parsing stream contents.", e);
+                    }
+
+                default:
+                    throw new InvalidOperationException("Error reading credentials from stream, 'type' field not specified.");
+            }
+        }
+
+        /// <summary> 
+        /// Helper method to return well known credential file path. This file is created by 'gcloud auth login'.
+        /// </summary>
+        private string GetWellKnownCredentialFilePath()
+        {
+            return Path.Combine(System.Environment.GetEnvironmentVariable(APPDATA_ENV_VAR),
+                CLOUDSDK_CONFIG_DIRECTORY,
+                WELL_KNOWN_CREDENTIALS_FILE);
+        }
+
+        /// <summary>
+        /// Helper method to load file stream.
+        /// </summary>
+        internal ICredential LoadCredentialFromFile(string credentialPath)
+        {
+            Logger.Info(String.Format("Loading Credential from file {0}", credentialPath));
+
+            using (Stream fs = GetStream(credentialPath))
+            {
+                return LoadFromStream(fs);
+            }
+        }
+
+        /// <summary>
+        /// Helper method to read the file stream.
+        /// </summary>
+        internal virtual Stream GetStream(string filePath)
+        {
+            return new FileStream(filePath, FileMode.Open);
+        }
+
+        /// <summary>
+        /// Helper method to get the environment variable.
+        /// </summary>
+        internal virtual string GetEnvironmentVariable(string variableName)
```

Comment:
Do we need it here?
This method is a little bit weird. You pass a parameter to the method, but never use it...
Why do we need it? Is it for mocking and testing only?